widgetssdk/src/main/java/com/glia/widgets/helper/Logger.kt

```diff
@@ -1,31 +1,96 @@
 package com.glia.widgets.helper
 
 import android.util.Log
+import com.glia.androidsdk.Glia
 import com.glia.widgets.BuildConfig
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
 
 object Logger {
+
+    private var coreSdkLogger: com.glia.androidsdk.internal.logger.Logger? = null
+    private var coreSdkLogDFunction: Method? = null
+    private var coreSDKLogEFunction: Method? = null
+
+    @Suppress("unused")
+    @JvmStatic
+    fun setCoreSdkLogger(coreSdkLogger: com.glia.androidsdk.internal.logger.Logger) {
+        d(TAG, "Core SDK logger set successfully")
+        this.coreSdkLogger = coreSdkLogger
+    }
+
+    @JvmStatic
+    fun getCoreSdkLogger() : com.glia.androidsdk.internal.logger.Logger? = coreSdkLogger
+
     @JvmStatic
     fun d(tag: String, message: String?) {
         if (BuildConfig.DEBUG) {
             // No need to log an empty message
-            Log.d(tag, message ?: return)
+            logDToCoreSdk(tag, message ?: return)
         }
     }
 
     @JvmStatic
     fun e(tag: String, message: String?) {
         if (BuildConfig.DEBUG) {
-            // //No need to log an empty message
-            Log.e(tag, message ?: return)
+            //No need to log an empty message
+            logEToCoreSdk(tag, message ?: return)
         }
     }
 
+    private fun logDToCoreSdk(tag: String, message: String?) {
+        if (!Glia.isInitialized()) return
+        if (message == null) return
+
+        if (coreSdkLogDFunction == null) {
+            try {
+                coreSdkLogDFunction = getCoreSdkLogger()?.javaClass?.getDeclaredMethod(
+                    "d",
+                    String::class.java,
+                    String::class.java
+                )
+            } catch (e: IllegalAccessException) {
+                Log.d(TAG, "Unable to log message using Core SDK logger $e")
+            } catch (e: InvocationTargetException) {
+                Log.d(TAG, "Unable to log message using Core SDK logger $e")
+            } catch (e: NoSuchMethodException) {
+                Log.d(TAG, "Unable to log message using Core SDK logger $e")
```

Comment:
Could you please explain the reason, why you made 3 different `catch` blocks with the same message?

Reply:
Thank you for pointing this out. Fixed. 888e2c058b34b48e89ec48bb086bc8ac1d2ce1a9